include/envoy/server/filter_config.h

```diff
@@ -116,6 +117,11 @@ class ServerFactoryContext : public virtual CommonFactoryContext {
    * @return the server-wide grpc context.
    */
   virtual Grpc::Context& grpcContext() PURE;
+
+  /** @return DrainManager& drain manager which has life no shorter than the life of the current
```

Comment:
This is the server context so by definition isn't the lifetime of the entire server?

Reply:
Theoretically yes. We seldom violates it... e.g. the initManager()

Reply:
Let me think about it if I could done better

Reply:
IIRC add this because I need to access the server drain manager. I didn't expect derived class returns a shorter life. Will update the comment.